module/zfs/zfs_sa.c

```diff
@@ -248,9 +248,21 @@ zfs_sa_set_xattr(znode_t *zp)
 	if (error) {
 		dmu_tx_abort(tx);
 	} else {
-		VERIFY0(sa_update(zp->z_sa_hdl, SA_ZPL_DXATTR(zfsvfs),
-		    obj, size, tx));
+		int count = 0;
+		sa_bulk_attr_t  bulk[2];
+		uint64_t ctime[2];
+
+		SA_ADD_BULK_ATTR(bulk, count, SA_ZPL_DXATTR(zfsvfs),
+		    NULL, obj, size);
+		/* Any xattr change is metadata change hence update the ctime */
+		zfs_tstamp_update_setup(zp, STATE_CHANGED, NULL, ctime);
+		SA_ADD_BULK_ATTR(bulk, count, SA_ZPL_CTIME(zfsvfs),
+		    NULL, &ctime, 16);
+		VERIFY0(sa_bulk_update(zp->z_sa_hdl, bulk, count, tx));
+
 		dmu_tx_commit(tx);
+		/* Update Inode ctime */
+		zfs_inode_update(zp);
```

Comment:
There's no need to call `zfs_inode_update()`.  The call to `zfs_tstamp_update_setup()` will update the ctime in the inode.  We're working to retire the need for `zfs_inode_update()` so the fewer callers the better.

Reply:
done